libnetdata/queue/queue.c

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#include "queue.h"
+
+struct nodes{
+	void *item;
+	struct nodes *next;			
+};
+
+struct queues{		
+	pthread_cond_t conde;
+	pthread_cond_t condf;
+	pthread_mutex_t lock;
+	node front;			
+	node rear;
+	int max;
+	int count;
+};
+
+int enqueue(queue q, void* i)	
+{		
+	node temp;
+	temp = (node) malloc(sizeof(struct nodes));
+	if (temp == NULL){
+		return 0;
+	}
+	pthread_mutex_lock(&q->lock);
+	while(q->count == q->max){
+		pthread_cond_wait(&q->condf, &q->lock);
+	}
+	temp->item = i;
+	temp->next = NULL;
+	if(q->count != 0)		
+	{		
+		q->rear->next = (node)temp;
+	}
+	else{
+		q->front = temp;
+	}
+	q->rear = temp;
+    	q->count++; 
+	pthread_cond_signal(&q->conde);
+	pthread_mutex_unlock(&q->lock);
+	return 1;
+}
+
+void* dequeue(queue q)			
+{			
+	void* temp = NULL;					
```

Comment:
This variable will point to the nodes' data that we will return. A better approach is to use the name of the field that will be assigned to it, eg. if this is `q->front->item` then it should be `item`, or if you follow my suggestion above it should be `q->front->data` and `data`.